### PR TITLE
Allow to change resolutions used in full-screen mode

### LINKED ISSUE #3505: Add resolution picker
It has been 7 years and we have somehow managed so far without adding an actual option for picking the resolution to run the game at! Weird and recent realization. Destination Sol does have a resolution picker so it might have some related code for inspiration - although it uses LibGDX so maybe not.

For Terasology usually the game just runs in windowed mode, but it is a plenty valid use case to switch the resolution used in full screen.

To complete: Add a new option under the video settings to pick from available resolutions in a drop-down. Apply the updated resolution via some sort of OpenGL/LWJGL magic (surely there'll be some handy howto examples on Google)

### PR DESCRIPTION
**Pull request by Edvinas01**
Adds resolution picker to video settings menu and resolves #3505. 

It can be found here:

The resolution picker works in the following way:
- By default the best matching resolution will be shown in the picker.
- After picking any resolution in windowed full-screen or windowed modes, it will only be persisted in `config.cfg` without actually changing the resolution.
- After picking any resolution in full-screen mode, game resolution will change to the selected one and will be persisted in `config.cfg` file.

### Outstanding before merging
When changing resolutions on Linux Mint 18 with dual-monitor setup, sometimes the game screen appears to be cut-off in the middle. This seems to happen based on which screen I start the game and occurs randomly.

### CODE REVIEW

**Comment by jellysnake** [comment]
Tested this out and it does indeed work.
However it's also very easy to accidentally select the wrong entry like I did and you end up with a wholly unusable screen.

I think we should add some form of timer, where after the timer the resolution changes revert, unless confirmed. This way if you accidentally select the wrong one you aren't stuck

**Comment by jellysnake** [comment]
@GooeyHub: Ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Edvinas01** [comment]
@jellysnake good point, I'll play around with this idea.

**Comment by Edvinas01** [comment]
Added a popup which will appear once resolution is changed in full-screen mode. If the timer runs out, resolution will be restored to the previous one:
</img>

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/engine/subsystem/lwjgl/LwjglResolution.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        this.displayMode = displayMode;
+    }
+
+    public DisplayMode getDisplayMode() {
+        return displayMode;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/engine/subsystem/lwjgl/LwjglResolution.java]: Oh noes! Minor code style violation: always use brackets. Easy to fix tho :-)

**Pull request merged**